.github/mergify.yml

```diff
@@ -12,6 +12,7 @@ pull_request_rules:
     - label!=do-not-merge
     - label!=needs-rebase
     - check-success=DCO
+    - label=e2e-success # this will wait for the e2e workflow to pass, this label will be set by the workflow
```

Comment:
This will have to now be conditional as well using the same conditions as what triggers the e2e workflow